src/coreclr/src/pal/src/safecrt/wcslwr_s.cpp

```diff
@@ -0,0 +1,43 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+/***
+*wcslwr_s.cpp - contains _wcslwr_s() routine
+*
+*
+*Purpose:
+*   _wcslwr_s converts, in place, any upper case letters in input string to
+*   lowercase.
+*
+*******************************************************************************/
+
+#include <wctype.h>
+#include <string.h>
+#include <errno.h>
+#include <limits.h>
+#include "internal_securecrt.h"
+
+#include "mbusafecrt_internal.h"
+
+DLLEXPORT errno_t __cdecl _wcslwr_s(char16_t *string, size_t sz)
+{
+    _VALIDATE_RETURN_ERRCODE(string != NULL, EINVAL);
+
+    char16_t *p = string;
+    size_t cur = sz;
+    while (cur-- > 0 && *p++ != 0);
+
+    if (cur == sz)
```

Comment:
This is not correct. The cur will not be equal to `sz` in case there was no terminating 0. It would be `(size_t)-1`, since it starts at sz and goes down.

Reply:
Was there any reason to not to use the PAL_wcsnlen? It would make this code easier to read.

Reply:
Ah true, it should have -1. I ran PAL tests and they were passing. I think there is no coverage for this error case. Should we add a positive and a negative `wcslwr_s` test in palsuite to exercise:

* assert that `wstring` with no null-terminator fails
* assert that `wstring` with one null-terminator in the middle gets all null after the first one in debug configuration

> Was there any reason to not to use the PAL_wcsnlen?

No reason, I saw in other places where `_RETURN_DEST_NOT_NULL_TERMINATED` case is handled with similar manual (bodied) loop. I will update this one.